includes/abstracts/abstract-wc-stripe-payment-gateway.php

```diff
@@ -1456,9 +1457,15 @@ function ( $item ) use ( $currency ) {
 				}
 				$product_description = substr( $item->get_name(), 0, 26 );
 				$quantity            = $item->get_quantity();
-				$unit_cost           = WC_Stripe_Helper::get_stripe_amount( ( $subtotal / $quantity ), $currency );
-				$tax_amount          = WC_Stripe_Helper::get_stripe_amount( $item->get_total_tax(), $currency );
-				$discount_amount     = WC_Stripe_Helper::get_stripe_amount( $subtotal - $item->get_total(), $currency );
+				if ( ! $quantity ) {
+					WC_Stripe_Logger::error(
+						"Stripe Level 3 data: Order item with ID {$item->get_id()} from order ID {$order_id} has no quantity set. Defaulting to 1."
+					);
+					$quantity = 1; // Default to 1 if quantity is not set.
```

Comment:
@wjrosa Thanks for adding the logs.

What do you think of not setting it to one and letting it error out? I wonder if we will be creating data integrity errors down the line. The real quantity could be greater than one, and due to some unknown issue, it's zero here. If we assume it's 1, we will be sending incorrect data to Stripe.

Reply:
Oh right. Do you mean closing this PR and leave it as-is, or adding an exception to be thrown on these cases? This could at least make our code safer against the fatals. I have added the exception for now. Let me know if you think that is a good idea 👍

Reply:
I was thinking of just keeping the logs without `quantity=1`, but adding an exception makes more sense along with the logs. Thanks Wesley!